flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/stopwithsavepoint/StopWithSavepointOperationHandlerImpl.java

```diff
@@ -217,6 +230,17 @@ public State onSavepointCreationFailure(Throwable throwable) {
             terminateExceptionally(throwable);
             return new FinalState();
         }
+
+        @Override
+        public State onAbort(Throwable cause) {
+            return onSavepointCreationFailure(cause);
```

Comment:
Why does `onAbort` trigger a savepoint creation failure? Couldn't it transition to `Finished` and fail the operation future?